### DISCUSSION TITLE
Pull organization docker image

### DISCUSSION THREAD

**Discussion by palu3492** [community/community#24927] [Actions]
<p>Hello,<br>
I’m hosting a Docker image on one of my organization’s GitHub repos. Right now, I’m using my personal account to pull the images, using my username and personal access token. Am I able to pull images using my organization instead of my personal account?</p>
<p>Edit: I read that some people use an alternate GitHub account just for doing this. Is that a reasonable approach?</p>

**Comment by airtower-luna** [comment]
<aside class="quote no-group" data-username="palu3492" data-post="1" data-topic="164411">
<div class="title">
<div class="quote-controls"></div>
 palu3492:</div>
<blockquote>
<p>Edit: I read that some people use an alternate GitHub account just for doing this. Is that a reasonable approach?</p>
</blockquote>
</aside>
<p>It’s described as one possible approach for SSH-based deployments in the documentation, see <a href="https://docs.github.com/en/developers/overview/managing-deploy-keys#machine-users">Machine users</a>. In principle the pros/cons listed there apply in your case, too, if you adjust for the different privileges.</p>
<p>The other important question is where you’re pulling to. For a server a machine user may make sense, for your own system I think a PAT of yours is fine.</p>

**Comment by jcansdale** [comment, marked as answer]
<p><a class="mention" href="/palu3492">@palu3492</a>,</p>
<aside class="quote no-group" data-username="palu3492" data-post="1" data-topic="164411">
<div class="title">
<div class="quote-controls"></div>
 palu3492:</div>
<blockquote>
<p>Am I able to pull images using my organization instead of my personal account?</p>
</blockquote>
</aside>
<p>Unfortunately not. You can only create PATs from a user account, not an org.</p>
<p>You can either use your own account or (as <a class="mention" href="/airtower-luna">@airtower-luna</a> mentioned), create a machine-user that is a member of your org. Alternatively, you can make the machine-user a collaborator on the repos it needs access to. Note, the PAT only needs the <code>read:packages</code> scope to access images associated with public <em>or</em> private repositories.</p>
<p>Another approach would be use the new GitHub Container Registry (<a href="http://ghcr.io" rel="noopener nofollow ugc">ghcr.io</a>). This could be used to publish public images (assuming that would be acceptable).</p>
<p>You can find more information here:</p><aside class="onebox githubfolder">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">
      <a href="https://docs.github.com/en/packages/guides/configuring-access-control-and-visibility-for-container-images#configuring-visibility-of-container-images-for-an-organization" target="_blank" rel="noopener nofollow ugc">docs.github.com</a>
  </header>
  <article class="onebox-body">
    

<h3><a href="https://docs.github.com/en/packages/guides/configuring-access-control-and-visibility-for-container-images#configuring-visibility-of-container-images-for-an-organization" target="_blank" rel="noopener nofollow ugc">Configuring access control and visibility for container images - GitHub Docs</a></h3>

<p><a href="https://docs.github.com/en/packages/guides/configuring-access-control-and-visibility-for-container-images#configuring-visibility-of-container-images-for-an-organization" target="_blank" rel="noopener nofollow ugc">//docs.github.com/en/packages/guides/configuring-access-control-and-visibility-for-container-images</a></p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>
